tools/check-regions-drift.py

```
@@ -0,0 +1,234 @@
+        with urllib.request.urlopen(request, timeout=60) as response:
+            return response.read().decode("utf-8")
+    except (urllib.error.URLError, OSError) as e:
+        raise CheckError(f"could not fetch {url}: {e}")
+
+
+def load_regions(text, source):
+    """Return {id: region} from a regions-vN.json envelope."""
+    try:
+        return {r["id"]: r for r in json.loads(text)["data"]["list"]}
+    except (ValueError, KeyError, TypeError) as e:
+        raise CheckError(f"{source} is not a regions directory envelope: {e}")
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)